### DISCUSSION TITLE
detectron2 in python 3.10

### DISCUSSION THREAD

**Discussion by ssoooong** [facebookresearch/detectron2#5119] [Q&A]
Hello.
I have a question about detectron2 version.

I understand that detectron2 supports up to torch 1.10.
Is there any way to get detectron2 from python 3.10?
(We estimate python 3.10 supports from torch 1.11)
(My Environment: CUDA version 11.1, python 3.10, GPU)

After searching, I found it difficult to use detectron2 in python 3.10, but I am asking if there is a way.

(case 1) Run well when python 3.8
pip 설치 torch==1.9.0+cu111 torchvison==0.10.0+cu111 -f https://download.pytorch.org/whl/torch_stable.html
pip install detectron2 -f https://dl.fbaipublicfiles.com/detectron2/wheels/cu111/torch1.9/index.html

(case 2) Python 3.10
Not executed at execution of the code above, tried running with a different version, but not executed
(Error when torch version is set to 1.11)

Thank you!

**Comment by greut** [comment]
Just build it yourself. Here is part of the Dockerfile I'm using to build it for amd64 and arm64 only +cpu but I guess it won't be hard to adapt to use cu111.

```
ENV PYTHON_VERSION=3.10
ENV CU_VERSION=cpu
ENV D2_VERSION_SUFFIX=+cpu
ENV PYTORCH_VERSION=1.12.1

# For not particular reasons, that one.
# https://github.com/facebookresearch/detectron2/commit/898507047cf441a1e4be7a729270961c401c4354
ARG DETECTRON2_COMMIT_ID=898507047cf441a1e4be7a729270961c401c4354  # A detectron2 version
RUN git clone --depth 100 --single-branch --branch main \
        https://github.com/facebookresearch/detectron2

WORKDIR /detectron2

RUN --mount=type=cache,target=/root/.cache/pip \
    PYTORCH_VERSION="${PYTORCH_VERSION}$([ "x86_64" = "$(uname -m)" ] && echo -n "${D2_VERSION_SUFFIX}" || echo -n "")" \
 && git reset --hard ${DETECTRON2_COMMIT_ID} \
 && pip install --progress-bar off \
        "torch==$PYTORCH_VERSION" \
        -f "https://download.pytorch.org/whl/${CU_VERSION}/torch_stable.html" \
 && python3 setup.py build \
 && python3 setup.py bdist_wheel -d "wheels/${CU_VERSION}"
```

**Comment by MiroPsota** [comment]
You can try my repository for building packages and PyPI simple index and see if it works for you:
https://github.com/facebookresearch/detectron2/discussions/5200

**Comment by gecjmanish** [comment]
@ssoooong do we have any solution for this, or still we need to wait for it;
@greut @MiroPsota @amyreese @ezyang

**Comment by CodePint** [comment]
Why was this never answered, python 3.10 is over 3 years old at this point?
Please create a wheel

**Comment by Gnoot01** [comment]
Bump

**Comment by gecjmanish** [comment]
I think most of people using cloud features
